Initial ExtraRepositories module

Cleptomania opened the pull request:
This creates a new module called ExtraRepositories. This largely serves to replace the WellKnownRepositories module, it serves the same purpose but is more configurable.

This allows the a space-separated string of keys to be passed in the `extraRepositories` field of `gradle.properties`. The keys placed in this field indicate which extra repositories will be added. Right now the three that were included by WellKnownRepositories are supported, and they are enabled by default when using this setting.

This does introduce a bit of a clash with WellKnownRepositories, so the functionality is currently that if you have that enabled, then this module will not be applied.

This is helpful because for example, when MCModDev goes down(and often in a way that is a hard crash when you check it, for example DNS resolution errors as of late), it makes it impossible to use mavenLocal, as the repositories added by GTNHGradle get searched before anything added by `repositories.gradle` in the project. This new setting enables developers to disable problematic repositories on a more individual basis.

eigenraven reviewed (changes requested):
This is a neat idea, but I'd rather have a setting `excludedWellKnownRepositories` that you can set to a list of repos to avoid enabling from the list in the orignal module, that way it doesn't duplicate the functionality with imo worse semantics

mitchej123 commented:
> This is a neat idea, but I'd rather have a setting `excludedWellKnownRepositories` that you can set to a list of repos to avoid enabling from the list in the orignal module, that way it doesn't duplicate the functionality with imo worse semantics

I like this, or just removing modded mc maven since it's been flakey for a few weeks now

Cleptomania commented:
I've changed this to use an `excludeWellKnownRepositories` setting which works similarly in that it is a space separated list of keys, but it now works such that it disables those repos from being added by `includeWellKnownRepositories`.

I've also went ahead and removed the MMD repo by default
